packages/samples/headless-react/src/components/quickview/quickview.class.tsx

```diff
@@ -0,0 +1,69 @@
+import {Component, ContextType} from 'react';
+import {
+  buildQuickview,
+  Quickview as HeadlessQuickview,
+  QuickviewState as HeadlessQuickviewState,
+  Unsubscribe,
+  Result,
+} from '@coveo/headless';
+import {AppContext} from '../../context/engine';
+
+interface QuickviewProps {
+  result: Result;
+}
+
+interface QuickviewState {
+  isModalOpen: boolean;
+  quickview: HeadlessQuickviewState;
+}
+
+export class Quickview extends Component<QuickviewProps, QuickviewState> {
+  static contextType = AppContext;
+  context!: ContextType<typeof AppContext>;
+
+  private controller!: HeadlessQuickview;
+  private unsubscribe: Unsubscribe = () => {};
+
+  componentDidMount() {
+    const result = this.props.result;
+    this.controller = buildQuickview(this.context.engine!, {
+      options: {result},
+    });
+
+    this.unsubscribe = this.controller.subscribe(() => this.updateState());
+  }
+
+  componentWillUnmount() {
+    this.unsubscribe();
+  }
+
+  private updateState() {
+    this.setState({quickview: this.controller.state});
+  }
+
+  private openModal() {
+    this.controller.fetchResultContent();
+    this.setState({isModalOpen: true});
+  }
+
+  private closeModal() {
+    this.setState({isModalOpen: false});
+  }
+
+  render() {
+    if (!this.state || !this.state.quickview.resultHasPreview) {
+      return null;
+    }
```

Comment:
I believe this could be shortened to `if (!this.state?.quickview?.resultHasPreview) { ... }`

Reply:
yep, indeed. Will make the change,